B9PartSwitch/PartSwitch/PartModifiers/ModuleDataHandlerBasic.cs

```diff
@@ -34,16 +34,54 @@ public ModuleDataHandlerBasic(PartModule module, ConfigNode originalNode, Config
         public override void OnWillBeCopiedActiveSubtype() => Deactivate();
         public override void OnWasCopiedActiveSubtype() => Activate();
 
-        private void Activate()
+        protected virtual void Activate()
         {
             module.Load(dataNode);
             module.Events.Send("ModuleDataChanged", moduleDataChangedEventDetails);
         }
 
-        private void Deactivate()
+        protected virtual void Deactivate()
         {
             module.Load(originalNode);
             module.Events.Send("ModuleDataChanged", moduleDataChangedEventDetails);
         }
     }
+
+    public class ModuleFuelTanksHandler : ModuleDataHandlerBasic
+    {
+        public ModuleFuelTanksHandler(
+            PartModule module, ConfigNode originalNode, ConfigNode dataNode, BaseEventDetails moduleDataChangedEventDetails
+        ) : base(module, originalNode, dataNode, moduleDataChangedEventDetails)
+        { }
+
+        protected override void Activate() => applyNode(dataNode);
+        protected override void Deactivate() => applyNode(originalNode);
+
+        private void applyNode(ConfigNode sourceNode) {
+            double volume = 0;
+            bool setsVolume = sourceNode.TryGetValue("volume", ref volume);
+            string type = null;
+            bool setsType = sourceNode.TryGetValue("type", ref type);
```

Comment:
Can we make `setsVolume` and `setsType` (and probably the values too) in the constructor rather than computing them here?  The original node will always have these fields so they'll end up getting hit even if the intention is not to modify one of them.

Reply:
Also I see we're throwing away everything on the node other than `volume` and `type`, do we really want that?